oak/server/rust/oak_runtime/src/node/wasm/mod.rs

```diff
@@ -421,6 +421,14 @@ impl WasmInterface {
     }
 
     /// Corresponds to the host ABI function [`wait_on_channels`](https://github.com/project-oak/oak/blob/master/docs/abi.md#wait_on_channels).
+    ///
+    /// May return the following error statuses:
+    /// - `Err(OakStatus::ErrTerminated)`: if [`Runtime`] is terminating. In this cases, read
+    ///   statuses in `status_buff` do not reflect the actual read statuses.
+    /// - `Err(OakStatus::ErrInvalidArgs)`: if cannot parse `status_buff` into channel handles.
+    /// - `Err(OakStatus::ErrBadHandle)`: if all channels have erroneous statuses.
```

Comment:
Just to double check (I know, I should know by now :P ): if there are 10 channels, and 9 of them are invalid and 1 ready, it returns Ok, but if all of them are invalid then it returns ErrBadHandle?

Reply:
That is true, and it sounds counterintuitive! IIRC the reason for adding the check for all invalid channels and then returning `Err(OakStatus::ErrBadHandle)` was to have an extra flag to let the caller functions more quickly detect the situation. But the caller functions may as well check the situation themselves.
That check can be removed as I have done in this [PR](https://github.com/project-oak/oak/pull/1123).